### PR TITLE
chain: Add Transaction::hash() method.

### PR DESCRIPTION
**Pull request by hdevalence**
This makes Transaction and Block have a consistent API.

Extracted from #1008.

### CODE REVIEW

**Review by dconnolly — DISMISSED** [review]
Any chance of a wee test to cover this?  🙏

**Comment by hdevalence** [comment]
Sure, I can add one, or more likely, change an existing one to cover this API.

**Comment by hdevalence** [comment]
@dconnolly we didn't have one, but I added one using the existing test vector.

I also filled in missing Display impls and tested those too.

File: zebra-chain/src/transaction/hash.rs [hunk 1]
```diff
@@ -9,16 +9,16 @@ use crate::serialization::{sha256d, SerializationError, ZcashSerialize};
 
 use super::Transaction;
 
-/// A hash of a `Transaction`
+/// A transaction hash.
 ///
-/// TODO: I'm pretty sure this is also a SHA256d hash but I haven't
-/// confirmed it yet.
+/// Note: Zebra displays transaction and block hashes in their actual byte-order,
```
Comment (dconnolly) [inline zebra-chain/src/transaction/hash.rs:14]: 👍

File: zebra-chain/src/transaction/tests/prop.rs [hunk 2]
```diff
@@ -14,6 +14,13 @@ proptest! {
         prop_assert_eq![tx, tx2];
     }
 
+    #[test]
+    fn transaction_hash_display_fromstr_roundtrip(hash in any::<Hash>()) {
```
Comment (dconnolly) [inline zebra-chain/src/transaction/tests/prop.rs:18]: 🕺

File: zebra-chain/src/transaction/tests/vectors.rs [hunk 3]
```diff
@@ -13,6 +13,20 @@ fn librustzcash_tx_deserialize_and_round_trip() {
     assert_eq!(&zebra_test::vectors::GENERIC_TESTNET_TX[..], &data2[..]);
 }
 
+#[test]
+fn librustzcash_tx_hash() {
```
Comment (dconnolly) [inline zebra-chain/src/transaction/tests/vectors.rs:17]: 👍

**Review by dconnolly — APPROVED** [review]
❤️✅

**Pull request merged**